llama_stack/distribution/utils/exec.py

```diff
@@ -92,7 +93,13 @@ def run_with_pty(command):
     if sys.platform.startswith("win"):
         return _run_with_pty_win(command)
     else:
```

Comment:
Oh I missed this change before. Why do we even `run` with interactive mode on in the first place? Isn't it something that is applicable for `build` (when no arguments passed?) only? Perhaps the right change is switching `run` to always use non-interactive?

Reply:
Yeah, I had to do this since it failed to run in the CI. It's a good point that `run` doesn't need it, I have no insight on why this is done in the first place. Maybe @ashwinb? Do you mind if we address this in a followup refactor? Thanks!

Reply:
I have to say that it seems wrong to change PTY-specific code path to ultimately fall back to non-PTY (the caller should pick the path depending on whether interactive mode is enabled). So I think this should change should be reverted.

Reply:
Sounds good. I’ll roll back all changes and run the server using python -m since I believe we should proceed with the original objective of the PR.